initrd/bin/cbfs-init

```diff
@@ -16,8 +16,15 @@ if [ -z "$CONFIG_PCR" ]; then
 	CONFIG_PCR=7
 fi
 
+if [ "$CONFIG_CBFS_VIA_FLASHROM" = "y" ]; then
+	# Use flashrom directly, because we don't have /tmp/config with params for flash.sh yet
+	/bin/flashprog -p internal --fmap -i COREBOOT -i FMAP -r /tmp/cbfs-init.rom \
+		&& CBFS_ARG=" -o /tmp/cbfs-init.rom" \
+		|| echo "Failed reading Heads configuration from flash! Some features may not be available."
+fi
+
 # Load individual files
-cbfsfiles=`cbfs -t 50 -l 2>/dev/null | grep "^heads/initrd/"`
+cbfsfiles=`cbfs -t 50 -l $CBFS_ARG 2>/dev/null | grep "^heads/initrd/"`
```

Comment:
@Tonux599 what happens when calling cbfs on MSI platform? What is the issue that needs this workaround?

Reply:
> Edit: and calling cbfs as normal results in an invalid header magic error (presumably because it's reading garbage).

Yes that's why I ask because novacustom boards have spi chips of 32mb and don't need that workaround.

Reply:
@Tonux599 that deserves a new issue to be tracked/fixed/resolved under Heads and then osresearch/flashtools. After which this thread can be resolved yes